### PR TITLE
add onramp.money

### PR DESCRIPTION
**Pull request by sajwanAshish**
## What is the purpose of the change

adds another option to buy tokens from (onramp.money)

### ClickUp Task

[ClickUp Task URL](PASTE_CLICKUP_TASK_URL_HERE)

## Brief Changelog
- when click on buy button, it add 1 more option called onramp.money.
- opens onramp.money on a iframe.
- users sees how much they are going to pay & how much coins are they coin to get.
- user address is prefilled
- user needs to pay & received token in their wallet

### CODE REVIEW

File: packages/web/localizations/en.json
Comment (jonator) [inline packages/web/localizations/en.json:1]: Reminder to run `yarn machine-translate` at the root to get other languages

**Comment by MaxMillington** [comment]
Looks like we might have some build failures due to type errors. Can you run `yarn build` locally and get it to pass? 

Screenshot 2023-12-12 at 11 26 38 AM

File: packages/web/localizations/en.json
Comment (sajwanAshish) [reply packages/web/localizations/en.json:1]: done

**Comment by sajwanAshish** [comment]
> Looks like we might have some build failures due to type errors. Can you run `yarn build` locally and get it to pass?
> 
> Screenshot 2023-12-12 at 11 26 38 AM

m using `yarn run generate` command to generate the `sprite-ids.ts` but its not working..

looks like sprite-ids.ts is build using `public/icons/sprite.svg` file - mentioned in `generate-sprite-ids.ts` file

how to i update `public/icons/sprite.svg` file ?

**Comment by MaxMillington** [comment]
> > Looks like we might have some build failures due to type errors. Can you run `yarn build` locally and get it to pass?
> > Screenshot 2023-12-12 at 11 26 38 AM
> 
> m using `yarn run generate` command to generate the `sprite-ids.ts` but its not working..
> 
> looks like sprite-ids.ts is build using `public/icons/sprite.svg` file - mentioned in `generate-sprite-ids.ts` file
> 
> how to i update `public/icons/sprite.svg` file ?

Sent you a slack

**Comment by sajwanAshish** [comment]
> > > Looks like we might have some build failures due to type errors. Can you run `yarn build` locally and get it to pass?
> > > Screenshot 2023-12-12 at 11 26 38 AM
> > 
> > 
> > m using `yarn run generate` command to generate the `sprite-ids.ts` but its not working..
> > looks like sprite-ids.ts is build using `public/icons/sprite.svg` file - mentioned in `generate-sprite-ids.ts` file
> > how to i update `public/icons/sprite.svg` file ?
> 
> Sent you a slack

fixed build issues

**Comment by CryptoAssassin1** [comment]
in logo ON text is only visible when you hover the mouse.
Screenshot from 2023-12-13 21-34-58
Screenshot from 2023-12-13 21-34-49

**Comment by kamal-sutra** [comment]
Says both Osmo and USDC supported but only Osmo is available currently. Shouldn't be a blocker if USDC is coming soon.

image
image

**Comment by kamal-sutra** [comment]
Couldn't link TG bot for a Portugal phone number.
Din't get OTP for an Indian phone number

image
image

**Comment by sajwanAshish** [comment]
> in logo ON text is only visible when you hover the mouse. Screenshot from 2023-12-13 21-34-58 Screenshot from 2023-12-13 21-34-49

logo updated

**Comment by arispech** [comment]
Osmo logo does not look good, it looks pixelated and unaligned.
Screenshot 2023-12-13 at 11 52 18 AM

**Comment by sajwanAshish** [comment]
> Couldn't link TG bot for a Portugal phone number. Din't get OTP for an Indian phone number
> 
> image image

Here is a dummy phone number you can use

Phone Number: +91 1122334455

OTP :  123456

**Comment by CryptoAssassin1** [comment]
There should be limit for integers so things dont overlap.
Screenshot from 2023-12-13 21-59-33

**Comment by kamal-sutra** [comment]
@sajwanAshish I tried with dummy number and scanned QR to pay with googlepay/phonepe with 200Rs.  dint went through. Is it because of dummy number or payments aren't open yet? 
We would like to test full version end to end with real phone numbers (EU, US, IND etc) and Tg accounts to make sure full integrations works well.

**Comment by sajwanAshish** [comment]
> There should be limit for integers so things dont overlap. Screenshot from 2023-12-13 21-59-33

this is sorted

**Comment by sajwanAshish** [comment]
> in logo ON text is only visible when you hover the mouse. Screenshot from 2023-12-13 21-34-58 Screenshot from 2023-12-13 21-34-49

sorted

**Review by DavideSegullo — APPROVED** [review]

**Comment by sajwanAshish** [comment]
i see there are conflicts in localizations files... shall i go ahead & fix conflicts or your team will take care of it

**Comment by DavideSegullo** [comment]
> i see there are conflicts in localizations files... shall i go ahead & fix conflicts or your team will take care of it
> 
> @DavideSegullo @kamal-sutra

You can fix them

**Comment by sajwanAshish** [comment]
> Osmo logo does not look good, it looks pixelated and unaligned. Screenshot 2023-12-13 at 11 52 18 AM

fixed

**Comment by MaxMillington** [comment]
@sajwanAshish looks like you just need to rebase against stage to resolve conflicts

**Comment by kamal-sutra** [comment]
Tried with Indian mobile number- KYC done in 5min.
Purchase with QR done in 5min
Sending osmo to my addy +5min
In 15min all done end to end.

**Review by CryptoAssassin1 — APPROVED** [review]

**Comment by kamal-sutra** [comment]
@sajwanAshish only inr fiat available for now? eur or usd?

**Review by jonator — CHANGES_REQUESTED** [review]
I got the following error when opening the modal from sidebar:

Screenshot 2023-12-21 at 12 17 56 PM

Also, I will say it was quite slow for me.

**Comment by MaxMillington** [comment]
@sajwanAshish any update on this?

**Comment by sajwanAshish** [comment]
> 123456

This test number has changed,

this is for Germany (EU) 
+49 1122334455 | 123456

for india plz use real numbers..
-- | --

**Comment by kamal-sutra** [comment]
@sajwanAshish it is not giving EUR option
image

**Comment by kamal-sutra** [comment]
> 1122334455

EUR option shows now, just couldn't test with dummy number, as it sends otp to that number

**Comment by sajwanAshish** [comment]
Pls try entering 123456 as otp

**Comment by kamal-sutra** [comment]
> > 1122334455

image

**Comment by sajwanAshish** [comment]
> > > 1122334455
> 
> image

plz try now

**Comment by kamal-sutra** [comment]
> > > > 1122334455
> > 
> > 
> > image
> 
> plz try now

got Fetching bank details for 5min and then this
image

**Comment by sajwanAshish** [comment]
> > > > > 1122334455
> > > 
> > > 
> > > image
> > 
> > 
> > plz try now
> 
> got Fetching bank details for 5min and then this image

Please try it again..

**Comment by kamal-sutra** [comment]
> > > > > > 1122334455
> > > > 
> > > > 
> > > > image
> > > 
> > > 
> > > plz try now
> > 
> > 
> > got Fetching bank details for 5min and then this image
> 
> Please try it again..

@sajwanAshish it is now giving me bank details. But i gave a dummy name as Recipient and i don't get the option to change to real name even after clearing cache. For now i will try with another wallet address

**Comment by kamal-sutra** [comment]
@sajwanAshish Even from another wallet it is not giving me option to write new Recipient name. I guess it is already linked to the dummy german number. Can u provide option to edit Recipient name please. Else bank transfer might get declined/lost.
image

**Comment by sajwanAshish** [comment]
> @sajwanAshish Even from another wallet it is not giving me option to write new Recipient name. I guess it is already linked to the dummy german number. Can u provide option to edit Recipient name please. Else bank transfer might get declined/lost. image

can we get on a call & sort out this out 

as you guys might want to test it for multiple countries before approving this change

**Comment by kamal-sutra** [comment]
> > @sajwanAshish Even from another wallet it is not giving me option to write new Recipient name. I guess it is already linked to the dummy german number. Can u provide option to edit Recipient name please. Else bank transfer might get declined/lost. image
> 
> can we get on a call & sort out this out
> 
> as you guys might want to test it for multiple countries before approving this change

Sure. can u send invite link to maquina.osmosis@gmail.com (but for tomorrow if possible, it's late night for me now)
cc @MaxMillington @CryptoAssassin1  if you wanna get on call

**Comment by kamal-sutra** [comment]
Purchase with EUR is successful. 
Suggestions- 
Allow 'space' in Recipient name field;
A check box would be helpful for the user to make sure the Name, Phone number etc are provided correctly.

Good to merge

**Review by kamal-sutra — APPROVED** [review]

**Review by jonator — APPROVED** [review]

**Review by CryptoAssassin1 — APPROVED** [review]

**Comment by kamal-sutra** [comment]
@sajwanAshish IMP
i just tried another transfer. i think there is a payment method network issue. 
This din't happened during QA few days ago. Alternative option to transfer by bank is also not available. Bank depsoit details are not given.
image

**Comment by sajwanAshish** [comment]
> @sajwanAshish IMP i just tried another transfer. i think there is a payment method network issue. This din't happened during QA few days ago. Alternative option to transfer by bank is also not available. Bank depsoit details are not given. image

should be working now..

**Comment by kamal-sutra** [comment]
> > @sajwanAshish IMP i just tried another transfer. i think there is a payment method network issue. This din't happened during QA few days ago. Alternative option to transfer by bank is also not available. Bank depsoit details are not given. image
> 
> should be working now..

Taking long time. Please check status
image

**Pull request merged**